### PR TITLE
Clarify PYTHONPATH changes and ``rootdir`` roles

### LINKED ISSUE #2589: Pytest adds parent instead of expected directory into python path. (shouldn't it add rootdir?)
Using pytest 3.1.3, python 2.7, OSX 10.12.5 

Looking at https://docs.pytest.org/en/latest/customize.html Which appears to be the most recent documentation.

Documentation says that it looks for pytest.ini to find rootdir. Instead it looks up from tests dir until it finds a dir without __init__.py. 

I'm not sure if this is a documentation error, a linking error in the docs (It's not pointing to the most recent version of the docs somehow) or an actual error in the code. I'm guessing it's just that the docs aren't up to date, but that's just a guess.

EDIT:

It appears that I asked the wrong question. (leaving it there for clarity)

The question isn't actually about rootdir. The issue is that the documentation doesn't clearly say what pytest puts in PYTHONPATH. As such, I assumed it put rootdir. And so I assumed it was not setting rootdir correctly. It turns out that it is setting rootdir just fine, but it doesn't use rootdir in pythonpath at all. 

I believe that there should be some (perhaps just some easier to find) documentation on how pytest sets pythonpath.

### PR DESCRIPTION
**Pull request by nicoddemus**
- Also minor adjustments in the docs (wording, formatting, links, etc).

Fix #2589

### CODE REVIEW

File: doc/en/customize.rst [hunk 1]
```diff
@@ -15,17 +15,31 @@ which were registered by installed plugins.
...
-printed as part of the pytest header.  The rootdir is used for constructing
-"nodeids" during collection and may also be used by plugins to store
-project/testrun-specific information.
+the existence of ``ini-files``.  The determined ``rootdir`` and ``ini-file`` are
+printed as part of the pytest header during startup.
+
+Here's a summary what ``pytest`` uses ``rootdir`` for:
+
+* Constructing ``nodeids`` during collection; each test is assigned
+  a unique ``nodeid`` which is rooted at the ``rootdir`` and takes in account full path,
```
Comment (The-Compiler) [inline doc/en/customize.rst]: nitpick: Why would `nodeid` and `ini-file` (both singular/plural) be monospaced? Those are "concepts", not variable names or code, right?

File: doc/en/example/index.rst [hunk 2]
```diff
@@ -1,7 +1,7 @@
 
 .. _examples:
 
-Usages and Examples
+Examples and customizations tricks
```
Comment (The-Compiler) [inline doc/en/example/index.rst]: nitpick: customization (singular)

File: doc/en/pythonpath.rst [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+Test modules / ``conftest.py`` files inside packages
+----------------------------------------------------
+
+Consider this file and directory layout::
+
+    root/
+    |- foo/
+       |- __init__.py
+       |- contest.py
```
Comment (The-Compiler) [inline doc/en/pythonpath.rst]: contest -> conftest?

File: doc/en/pythonpath.rst [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+:ref:`test discovery`.
+
+Standalone test modules / ``conftest.py`` files
+-----------------------------------------------
+
+Consider this file and directory layout::
+
+    root/
+    |- foo/
+       |- contest.py
```
Comment (The-Compiler) [inline doc/en/pythonpath.rst]: contest -> conftest?

**Comment by The-Compiler** [comment]
Also cc @pfctdayelise @hackebrot who probably have some more experience with writing/reviewing documentation than I do.

File: doc/en/customize.rst (hunk 1 above)
Comment (nicoddemus) [inline doc/en/customize.rst]: You are right, consistency is important (not nitpick at all IMHO) 😉

File: doc/en/example/index.rst (hunk 2 above)
Comment (nicoddemus) [inline doc/en/example/index.rst]: 👍

File: doc/en/pythonpath.rst (hunk 3 above)
Comment (nicoddemus) [inline doc/en/pythonpath.rst]: Fixed

File: doc/en/pythonpath.rst (hunk 4 above)
Comment (nicoddemus) [inline doc/en/pythonpath.rst]: Fixed

**Comment by nicoddemus** [comment]
Done all changes requested by @The-Compiler, thanks!

**Comment by nicoddemus** [comment]
@The-Compiler anything else you would like me to change? If not, could you update the status to approve? 😉 

Gentle ping @hackebrot @pfctdayelise

**Comment by The-Compiler** [comment]
I think this is clearly an improvement over the status quo, so let's merge it :wink: Thanks!

**Comment by nicoddemus** [comment]
Thanks!

**Comment by hackebrot** [comment]
Hey @nicoddemus @The-Compiler! 👋 

Sorry for the late response. I started my review right when you merged this PR. 😁 I'll submit a new PR with any changes that I was about to suggest, so we can discuss there. 👍

**Pull request merged**